app/src/main/java/com/cooper/wheellog/MainActivity.java

```diff
@@ -1417,8 +1453,7 @@ private void logEvent(String message) {
         if (eventsTextView == null) {
             return;
         }
-        SimpleDateFormat formatter = new SimpleDateFormat("HH:mm:ss", Locale.getDefault());
-        String formattedMessage = String.format("[%s] %s\n", formatter.format(new Date()), message);
+        String formattedMessage = String.format("[%s] %s\n", timeFormatter.format(new Date()), message);
```

Comment:
I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [FS: Format string should use %n rather than n (VA_FORMAT_STRING_USES_NEWLINE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#fs-format-string-should-use-n-rather-than-n-va-format-string-uses-newline).
This format string includes a newline character (\n). In format strings, it is generally  preferable to use %n, which will produce the platform-specific line separator.